scripts/rebalance_subject_split.py

```
@@ -0,0 +1,283 @@
+def parse_targets(raw_targets: list[str]) -> dict[str, float]:
+    """
+    Builds the target fraction mapping by applying label=fraction overrides to the default targets.
+    
+    Parameters:
+        raw_targets (list[str]): Iterable of override strings in the form "label=fraction". Each override updates or adds an entry in the returned mapping.
+    
+    Returns:
+        dict[str, float]: Mapping from label to target fraction (each value between 0.0 and 1.0 inclusive).
+    
+    Raises:
+        ValueError: If an override is malformed (missing '='), has an empty label, contains a non-numeric fraction, or if the fraction is outside the range 0.0–1.0.
```

Ruff (0.15.2)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)